src/OpenTelemetry.Instrumentation.AspNetCore/TracerProviderBuilderExtensions.cs

```diff
@@ -51,22 +55,44 @@ public static class TracerProviderBuilderExtensions
 
         internal static TracerProviderBuilder AddAspNetCoreInstrumentation(
             this TracerProviderBuilder builder,
-            AspNetCoreInstrumentation instrumentation)
+            AspNetCoreInstrumentation instrumentation,
+            IServiceProvider serviceProvider = null)
         {
+            // For .NET7.0 onwards activity will be created using activitySource.
+            // https://github.com/dotnet/aspnetcore/blob/bf3352f2422bf16fa3ca49021f0e31961ce525eb/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L327
+            // For .NET6.0 and below, we will continue to use legacy way.
+#if NET7_0_OR_GREATER
+            // TODO: Check with .NET team to see if this can be prevented
+            // as this allows user to override the ActivitySource.
```

Comment:
would be good to write a unittest which simulates this.

Reply:
Added